src/mesh/mesh.cpp

```diff
@@ -1213,10 +1233,32 @@ std::shared_ptr<MeshBlock> Mesh::FindMeshBlock(int tgid) const {
 bool Mesh::SetBlockSizeAndBoundaries(LogicalLocation loc, RegionSize &block_size,
                                      BoundaryFlag *block_bcs) {
   bool valid_region = true;
-  block_size = forest.GetBlockDomain(loc);
-  auto bcs = forest.GetBlockBCs(loc);
-  for (int i = 0; i < BOUNDARY_NFACES; ++i)
-    block_bcs[i] = bcs[i];
+  block_size = GetBlockSize(loc);
+  if (loc.tree() >= 0) {
+    auto bcs = forest.GetBlockBCs(loc);
+    for (int i = 0; i < BOUNDARY_NFACES; ++i) {
+      block_bcs[i] = bcs[i];
+    }
+    return valid_region;
+  }
+
+  for (auto &dir : {X1DIR, X2DIR, X3DIR}) {
+    if (!block_size.symmetry(dir)) {
+      std::int64_t nrbx_ll = nrbx[dir - 1] << (loc.level() - root_level);
+      if (loc.level() < root_level) {
+        std::int64_t fac = 1 << (root_level - loc.level());
+        nrbx_ll = nrbx[dir - 1] / fac + (nrbx[dir - 1] % fac != 0);
+      }
+      block_bcs[GetInnerBoundaryFace(dir)] =
+          loc.l(dir - 1) == 0 ? mesh_bcs[GetInnerBoundaryFace(dir)] : BoundaryFlag::block;
+      block_bcs[GetOuterBoundaryFace(dir)] = loc.l(dir - 1) == nrbx_ll - 1
+                                                 ? mesh_bcs[GetOuterBoundaryFace(dir)]
+                                                 : BoundaryFlag::block;
+    } else {
+      block_bcs[GetInnerBoundaryFace(dir)] = mesh_bcs[GetInnerBoundaryFace(dir)];
+      block_bcs[GetOuterBoundaryFace(dir)] = mesh_bcs[GetOuterBoundaryFace(dir)];
+    }
+  }
```

Comment:
Are these changes due to a merge issue? They look like old code that got removed in one of the forest PRs.

Reply:
Yeah I agree I don't know how this old code got back into this merge, I mistakenly thought it was part of the fix for periodic vs block BCs but it isn't I don't think. `Mesh::SetBlockSizeAndBoundaries` should be unchanged now, thanks for noticing this.

Reply:
yeah, weird. Sometimes git does strange things...